src/Security/LogoutSuccessHandler.php

```diff
@@ -51,14 +45,10 @@ public function onLogoutSuccess(Request $request): RedirectResponse
     {
         $session = $request->getSession();
 
-        if ($session && $session->has('_contao_logout_target')) {
-            return new RedirectResponse($session->get('_contao_logout_target'));
-        }
-
-        if ($this->scopeMatcher->isBackendRequest($request)) {
-            return new RedirectResponse($this->router->generate('contao_backend_login'));
+        if (null !== $session && $session->has('_contao_logout_target')) {
+            return $this->httpUtils->createRedirectResponse($request, $session->get('_contao_logout_target'));
         }
 
-        return new RedirectResponse($this->router->generate('contao_root'));
+        return $this->httpUtils->createRedirectResponse($request, 'contao_root');
```

Comment:
Backend logout to `contao_root` seems totally wrong to me.

Reply:
The backend logout does not need a custom success handler, this one is only for the frontend. We know the redirect target so we can simply use the configuration (see https://github.com/contao/manager-bundle/commit/a0f96f13a34419bf0068fe4618444c0e961edda3):